core/src/ballot_box.rs

```diff
@@ -463,17 +479,23 @@ impl BallotBox {
 
         let finalized_ballot = Ballot::new(meta_merkle_root);
 
-        // Check that the merkle root is one of the existing options
-        if !self.has_ballot(&finalized_ballot) {
-            return Err(TipRouterError::TieBreakerNotInPriorVotes);
-        }
+        let finalized_ballot_tally = self
+            .ballot_tallies
+            .iter()
+            .find(|t| t.ballot.eq(&finalized_ballot))
+            .ok_or(TipRouterError::TieBreakerNotInPriorVotes)?;
 
-        self.set_winning_ballot(finalized_ballot);
-        Ok(())
-    }
+        //TODO should we change the tallies?
 
-    pub fn has_ballot(&self, ballot: &Ballot) -> bool {
-        self.ballot_tallies.iter().any(|t| t.ballot.eq(ballot))
+        // // Check that the merkle root is one of the existing options
+        // // TODO do we want to be able to set a tie breaker that wasn't voted on?
+        // // What if there were malicious merkleroots?
+        // if !self.has_ballot(&finalized_ballot) {
+        //     return Err(TipRouterError::TieBreakerNotInPriorVotes);
+        // }
+
+        self.set_winning_ballot_tally(*finalized_ballot_tally);
```

Comment:
what's the reasoning for changing to use this method?

Reply:
Changed back